### DISCUSSION TITLE
Issues with multi-episode files

### DISCUSSION THREAD

**Discussion by Bekreth** [jellyfin/jellyfin#11865] [Q&A] [answered]
I've been using Jellyfin for a while as a Movie and Music server without issue, but am really struggling to get TV Shows to work properly.  I've through the `Shows` section a number of times in the docs (https://jellyfin.org/docs/general/server/media/shows) and believe I'm following the instructions but am still running into an issue where files that contain multiple episodes only show up as one episode. Pulling the path directly from Jellyfin looks like so:
* `/media/Shows/Spice and Wolf (2009)/Season 02/Spice and Wolf S02E00-E06.mp4`
* `/media/Shows/Spice and Wolf (2009)/Season 02/Spice and Wolf S02E07-E12.mp4`

But I only ever see 1 "episode" per disc (each being 2+ hours long). I've attempted to use several different plugins for metadata fetching without success:
* aniDB
* AniList
* Kitsu
* tvDB
* TheMovieDB
* The Open Movie Database

I assume that this is user error, but could use some help spotting what specifically I'm goofing up here. I'm using Jellyfin 10.9.2

**Comment by Burner1515** [comment]
I might be understanding this incorrectly, but TV Shows should have each episode in a separate file. For Spice and Wolf one possible source for episodes is

https://www.themoviedb.org/tv/26707-spice-and-wolf

Since you said Disc, I am assuming you mean BluRay Order (DVD). That would be this show order for season 2

https://www.themoviedb.org/tv/26707-spice-and-wolf/episode_group/5ff1bca766e469003ec05766/group/607b9bebd8f44e002a8283d0

so you would want to have each episode in it's own file like

/media/Shows/Spice and Wolf (2009)/Season 02/Spice and Wolf S02E01.mp4
~
/media/Shows/Spice and Wolf (2009)/Season 02/Spice and Wolf S02E13.mp4

In Jellyfin you would set the metadata to pull from themoviedb and for order choose blu-ray. You will need to refresh the metadata, but then it should be as themoviedb has it for BluRay Order (DVD).

That's just an example, you could use any metadata source that works as long as the files you have match the episode list the metadata source is using for the season. Like themoviedb also has Original Air Date and Funimation Order (Digital). if you set your Jellyfin to one of those for this show, you need the episodes to match the season and order listed.

This one looks simple and the same for each, but there are other Anime shows that will have air date order be 5 seasons while disc order is 8 seasons (EX Naruto). Once you figure out what order your files are set as a TV show renaming tool can help out a lot.

**Reply by Bekreth** [reply]
> so you would want to have each episode in it's own file

I had assumed it was possible for multiple episodes to be part of a single file due to the example tree in the https://jellyfin.org/docs/general/server/media/shows/

```
Shows
├── Series Name A (2010)
│   ├── Season 00
│   │   ├── Some Special.mkv
│   │   ├── Series Name A S00E01.mkv
│   │   └── Series Name A S00E02.mkv
│   ├── Season 01
│   │   ├── Series Name A S01E01-E02.mkv    <------- This guy here
│   │   ├── Series Name A S01E03.mkv
│   │   └── Series Name A S01E04.mkv
```

Have I misunderstood this? Or is there possibly an error in the documentation?  I do have the DVDs for this show, and I think I could rip it apart into an episode a file with ffmpeg, but if its possible for a single file to house multiple episodes, it would make it quite a bit easier.

**Reply by Burner1515** [reply, marked as answer]
> │   ├── Season 01
> │   │   ├── Series Name A S01E01-E02.mkv    <------- This guy here

I thought that was for shows where the episodes are considered multiple for their time slot or a special that's considered two episodes in one. Looking for examples of the former, like Adventure Time, TMDB had them all as separate episodes, even though the original time slot had two 10 minute stories that could be considered one episode. The latter it would be a 'to be continued' type with the first part week 1 and the exciting conclusion week 2. The examples I found of that are Speed Racer, and TMDB still put the episodes in order with each part being it's own episode. I took a look at TVDB and speed racer season one has 52 episodes same as TMDB.

I did find a couple threads talking about this.

https://forum.jellyfin.org/t-multi-part-episode-naming-not-working

https://www.reddit.com/r/jellyfin/comments/igx3qh/how_to_organise_files_when_multiple_episodes_are/?utm_medium=android_app&utm_source=share

so one notes it's by design, the other says this should work if you have the chapters correct.

You can try the chapters thing, but to me the bottom line is you are pulling your data from these metadata sources and jellyfin is trying to apply your files to the source. If the source has all the episodes broken up, you would want to break them up to match or find a source that has the seasons and episodes in the same order you do.

One last thing I noticed is the https://jellyfin.org/docs/general/server/media/shows/ guide barely mentions the E01-E02 part. In fact, it's just seen in the folder structure example only. The guide goes into much more detail on episodes split across multiple parts so that seems to be most common. An example would be if each episode of spice and wolf was ripped by chapters, and each episode had 2-3 chapters. You would append each chapter for a single episode with Part x and jellyfin would combine the files as the single episode in the GUI.

**Reply by Shadowghost** [reply]
Multi-episode files will be represented as a single episode within the season.
What is shown for this episode as metadata depends on the provider. E.g. TMDB should concat the titles and descriptions of all the included episodes (I validated this recently and it is working).

**Reply by Bekreth** [reply]
Thank you for the help! It'll be easy enough to switch to having my file split up by episodes based on chapters.

**Reply by gnc-07** [reply]
> Thank you for the help! It'll be easy enough to switch to having my file split up by episodes based on chapters.

If splitting up by episodes based on chapters worked, how did you name the file? Could you provide an example? Because I am having similar problems so it would be helpful to know what my naming scheme should be. Thanks in advance!

**Comment by tmulle** [comment]
I am running into this with `The Office - Season 5` where there are two episodes that are two parts in one file from the Blu Ray Box set.

S05E14-E15 and S05E16-17 but the meta provider sites seem to have them as separate episodes.

I also found this with my Friends Blu Ray collection also so I'd like to fix that as well.

I've been using the naming convention as shown above because it was mentioned in the docs. But it looks like it ultimately is up to the meta providers?

How would I go about splitting the file apart to match the metaproviders? I'm using MakeMKV to rip my Blu-Rays.

**Reply by TheMelmacian** [reply]
The whole point of the SxxEyy-Ezz is to tell jelyfin (the metadata provider plugin) that he file contains a range of episodes.
If the plugin supports this, it will fetch the title and description of all included episodes.
Since they are separate episodes they are correctly listed as such by the provider.

If you prefer to have separate files for each episode you can use [MKVToolNix](https://mkvtoolnix.download/) to split the file (without reencoding) based on chapter marks:
- open mkv in MKVToolNix GUI
- open Output tab
- Split mode: Before Chapter
- add chapter number(s) at which the file should be split (comma separated)
- set destination path and file name
- Start multiplexing

**Reply by tmulle** [reply]
Thanks for the info, so I might be doing things wrong when I'm doing my TV shows. Movies I had working perfectly.

I have been using this site for the episode listings: https://theoffice.fandom.com/wiki/Season_5 and manually numbering the files based on the information on the listing for each episode. So the first episode says it's S05E01-S05E02 and that's what I numbered the files.

But I see on https://www.themoviedb.org/tv/2316-the-office/season/5/episode/1 that all the episodes are numbered individually.

So, am I correct in thinking I should just be using the `themoviedb` numbering scheme and forget about using the wiki site? 

So, my S05E01-E02 because just becomes S05E01 (Weigh Loss) and what I have labelled as S05E03 (Business Ethics) becomes S05E02 (Business Ethics) based on the `themoviedb`?

**Reply by TheMelmacian** [reply]
Depends. If it's a double episode and the Blu-ray combines them, then the file is correctly named.
I have shows where double episodes were combined seamlessly for one season but not for the other. But it should be quite easy to check if a file contains one or more episodes (runtime, file size etc.).

But yes, if you want jellyfin to automatically fetch the correct metadata, you should name the episodes according to the selected providers. Cause jellyfin will fetch the metadata from the providers and not some fandom wiki ;)

Be aware that for some shows different orderings exist. For example Aired Order vs DVD Order.
Sometimes episodes are aired out of order but are correct on the DVD/Blu-ray etc.
The order can be selected on the metadata editor.

For example S05E01 of The Office just has a longer runtime in the aired and dvd order:
https://thetvdb.com/series/the-office-us/seasons/official/5
https://thetvdb.com/series/the-office-us/seasons/dvd/5

but the alternate order splits the longer episodes in two, and is the same as on the fandom wiki:
https://thetvdb.com/series/the-office-us/seasons/alternate/5


I use thetvdb.com as primary provider for shows. Especially because of the support for many alternate orderings.


So
- check your files after ripping is complete
- check ordering on metadata provider (thetvdb.org)
- if you prefer individual episodes use MkvToolNix to split files based on chapter marks
- name files accordingly
- select matching ordering in jellyfin if it differs from standard order

**Reply by tmulle** [reply]
Thanks for the good info and I'm understanding things better for TV shows.

I am using `the movie db` as the default for the tv shows and for the most part it works fine, but for my Friends episodes 14-15, they are a part 1 and 2. But according to this https://www.themoviedb.org/tv/1668-friends/season/6/episode/15 they have the episode as 45 mins but when I ripped the Blu Ray it is two separate files which I labeled `S05E15` and `S05E16` but that messes up episode 16 which is actually the one called `The One with Unagi` and not `part 2 of the one that could've been`

Is the entry on `The Movie DB` incorrect? `TheTVDB` has it listed as `S06E15` two times, assuming a `part1` and `part2` https://thetvdb.com/series/friends/seasons/dvd/6

So far `TheMovieDB` has been good, but things like this cause confusion among the different meta providers and I keep hearing both good and bad things about `TheTVDB` because it is now subscription based and a plugin that needs to be added.

Can I correct my file names to work with the standard `The Movie Database` or am I out of luck? 

I see a few options:

1. Fix the naming somehow to work with The Movie DB (if possible)
2. Fix the meta on that site to match the others
3. Merge the two files together into on large file
4. Switch to TVDB which I guess means purchase a yearly subscription so Jellyfin can pull data?
5. Use another free alternative? Which one?

Thanks for all your help, I'm new to the world of tv series ripping and it's been quite eye opening and confusing at the same time with all the differences between Air Date, DVD, Alternative, Absolute ordering, etc.

**Reply by TheMelmacian** [reply]
As far as I know the thetvdb-plugin for jellyfin has an API-token included, so no subscription necessary for jellyfin users.
But I don't know if it stays that way.

To make your files work with TheMovieDB either use parts `S06E15 part1`, `S06E15 part2` or combine them with MkvToolNix (open first part, drag and drop second part and select "add additional part to existing source file", define output path and filename, start multiplexing).

**Reply by tmulle** [reply]
Ah ok, so I installed the TVDb plugin and it appears to be working without any other configuration. 
So, I switched my tv shows to use that provider as the primary and theMovieDb secondary.

I am finding that it's best to use `Aired` ordering instead of DVD because it isn't picking up `part 1` `part 2` namings, so using the aired order format seems to make everything a single episode number which looks like it is correctly labelling my entries.

**Reply by TheMelmacian** [reply]
Aired and DVD order on thetvdb for The Office Season 5 are the same. It was the alternate order that lists double episodes as two separate entries ;)
But glad to hear that everything works now.

**Reply by tmulle** [reply]
You're right. I just saw on the TvDB website they say for the `BluRay` ordering which I have to use the `Alternate DVD` ordering. So I tried that in JF and it seems to be working for my office season 6 episodes.

Guess I'll keep that option and see how it goes
